### ISSUE TITLE
Change in time segments with masking

### ISSUE THREAD

**Issue by EEPuckett** [stschiff/msmc-tools#46]
Dear Dr. Schiffels and other msmc users,

I'm trying to better understand how msmc works.  The figure below is a single sample (2 haplotypes; phased) and I've run through msmc2.  The black line is the result using only the mask from bamCaller.  The purple line has an additional mask on the genes and flank regions; the pink line has the mappability mask from SNPable added.  I used the default time segments.

The results with the additional masking push the left time boundary far beyond the speciation age of my study organism.  I'm trying to determine if I have a problem with this analysis as I did not expect either masking file to change the results in this way.

I appreciate any insight.
Emily

![psmcprime](https://user-images.githubusercontent.com/46551580/187719942-3c381bd7-69d1-4f4a-8409-d8e60b5ad882.jpeg)

**Comment by stschiff** [comment]
OK, this is complicated. MSMC determines the time-boundaries from the average heterozygosity of the input data. Of course, with different masks you may get different thetas, so time will be scaled differently. Not sure if I read the Log-scale correctly, but it looks like you get a factor of 2 or so between the end point of the different curves, which is not really expected. You can check the different thetas in the log file, and perhaps confirm with an external tool from the input files.

Basically you need to investigate why these masks result in different thetas.

**Comment by EEPuckett** [comment]
I think I have figured this out.  While there was a problem with my mappability mask (related to only running the first segment coming out of splitfa for snpable; thus, only having a positive mask for a fraction of each chr), the real issue was with phasing.

My goal is to run msmc to get input for msmc-IM between two species that have signals of introgression.  When I phased all of the samples (beagle v5) from both species together, that moves the time bounds to the right (ie- back in time).  However, when I phase the species separately, I see the time bounds as I expect based on published phylogenies.

**Comment by stschiff** [comment]
Ah OK. So incorrect phasing may have moved mutations in biased ways onto haplotypes, so that some haplotypes were much more similar than expected. Interesting, but not too surprising given the complexity of statistical phasing
